### DISCUSSION TITLE
How to automatically parse the numeric value for a column when the file is opened?

### DISCUSSION THREAD

**Discussion by lulunac27a** [saulpw/visidata#2572] [Q&A]
I want to see the file contents with columns with only the numeric values in all rows should display the numeric values automatically instead of manually type `#` for integer values or `%` for floating point values. How can I do this?

**Comment by frosencrantz** [comment]
This thread might have information that is useful: https://github.com/saulpw/visidata/discussions/1488 

In that thread look for  `knownCols` and `afterLoad`.  Also mentioned in that thread, look for "autotune" and [skeept](https://github.com/skeept)'s `guessType` command.

**Reply by lulunac27a** [reply]
When I open the file with numeric values, the columns that have numeric values should convert from string to numeric values automatically instead of manually converting the column formatting type.

**Reply by saulpw** [reply]
@lulunac27a You might want to use Microsoft Excel, which works as you expect: https://www.howtogeek.com/816620/microsoft-is-finally-fixing-excels-data-conversion-problem/

**Reply by lulunac27a** [reply]
What about other file types like CSV, JSON, and TSV?
